src/App.jsx

```diff
@@ -1,35 +1,35 @@
-import { useState } from 'react';
+import { useContext, useState } from 'react';
+import { ThemeProvider } from 'styled-components';
 import AppBar from './components/app-bar/AppBar';
+import AppContext from './context/app-context';
+import GlobalStyles from './global-styles';
 import DetailsView from './pages/DetailsView';
 import HomeView from './pages/HomeView';
+import { darkTheme, lightTheme } from './styles/themes';
+import { types } from './types/types';
 
 const App = () => {
-  const [searchValue, setSearchValue] = useState('react');
   const [selectedVideo, setSelectedVideo] = useState(null);
-  const [page, setPage] = useState('home');
-
-  const searchHandler = (value) => {
-    if (page !== 'home') {
-      setPage('home');
-    }
-    setSearchValue(value);
-  };
+  const { page, changePage, appTheme } = useContext(AppContext);
```

Comment:
**Optional**: 

You can abstract `AppContext` in the `app-context.js` file so that file exposes something like a `useAppContext` so that you you can use that hook directly in other files instead of `useContext` with `AppContext` as a param.

Reply:
Ok, i will try to implement it. Thank you so much!!

And thank you for your comments, it fills me with satisfaction to know that you are enjoying what I am doing so far.